sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/Triggers.scala

```diff
@@ -27,3 +32,34 @@ import org.apache.spark.sql.streaming.Trigger
 @Experimental
 @Evolving
 case object OneTimeTrigger extends Trigger
+
+/**
+ * A [[Trigger]] that runs a query periodically based on the processing time. If `interval` is 0,
+ * the query will run as fast as possible.
+ */
+@Evolving
+private[sql] case class ProcessingTimeTrigger(intervalMs: Long) extends Trigger {
```

Comment:
Could you put this into another file like `ContinuousTrigger.scala` and the previous `ProcessingTime`?
If possible, please do `git mv` for renaming `ProcessingTime` to `ProcessingTimeTrigger`. Then, update the new file.

Reply:
I moved this as @srowen's suggestion (https://github.com/apache/spark/pull/24996#discussion_r299221473) as OneTimeTrigger is there without its own file.

I'm still not sure, but if the intention of deprecation is hiding implementations to end users, actually I'd also like to move ContinuousTrigger to Triggers.scala, as they can be controlled together.

If we change the mind to have file per implementation, Triggers.scala would be better to be renamed as OneTimeTrigger.scala too.

Reply:
Ya. `Moving ContinuousTrigger to Triggers.scala` is also a possible way to be consistent.

Reply:
We might need to keep original class as we haven't deprecated it yet, and to allow end users to only create Trigger implementations as `Trigger.xxx`, we may also want to deprecated some more classes as well.

The change may look like below commit:
https://github.com/HeartSaVioR/spark/commit/f8488cfb0a045806e125215b933fc10931b1b431

IMHO this could be considered as another issue as more deprecations are happening. WDYT?

Reply:
Ur, moving looks okay, but the new deprecation of `OneTimeTrigger` is unexpected one for me.
```
@deprecated("use Trigger.Once()", "3.0.0")
// NOTE: In later release, we can change this to `private[sql]` and remove deprecated.
case object OneTimeTrigger extends Trigger
```
Please make another PR for the deprecation of `OneTimeTrigger` if we really need that.

If the PR has multiple themes unexpectedly, we cannot merge it.

Reply:
OK I'll make another PR for deprecation of OneTimeTrigger, and change the topic of this PR to "places all Trigger implementations to Triggers.scala". Just pushed the new commit.

Reply:
I think you are moving the opposite direction from the way I recommended. Now, you add a new deprecate in this PR.
```
@deprecated("use Trigger.Continuous(intervalMs)", "3.0.0")
// NOTE: In later release, we can remove this.
```

Why do you add that here? I'm -1 for this PR.

Reply:
First of all, you recommended opposite direction what I got from @srowen, as I commented earlier, so I just would like to keep this as it is. Let's avoid contributors to do back and forth because of conflict between reviewers.

Second, the new commit reflects your comment like above:

> Moving ContinuousTrigger to Triggers.scala is also a possible way to be consistent.

and I also provided commit and ask for confirmation, and you confirmed.

https://github.com/apache/spark/pull/24996#discussion_r300817205

In confirmation comment I guess you were only concerning changing OneTimeTrigger together, so I excluded it.

Did I miss something?

Reply:
Got it. Please follow PMC's opinion, @HeartSaVioR . Sorry if you feel uncomfortable. I wanted to help and merge this PR, but it seems that it doesn't help you. I'll leave this PR to @srowen fully.

Reply:
@dongjoon-hyun I don't particularly care where the implementations live, I'm just looking for consistency. It's not terribly consistent to begin with. I think putting the private implementations in `Triggers.scala` is a step in the right direction, given that this file exists at all. I think this change is OK as-is. I don't think we have to move other implementations, though, anything that improves consistency is OK.

Reply:
Building on the comment I made in a different [thread](https://github.com/apache/spark/pull/24996/files#r299514393), I agree things were not very consistent. Since we are refactoring this, lets make sure things move in the right direction and improve consistency. My opinion on where we should ideally be is 
- Minimize number of public classes that the user needs to know about. That means that ideally there NOT should be multiple public classes and ways for specifying a type of trigger. So there is should be only `Trigger.ProcessingTime` and `ProcessingTimeTrigger` should be hidden.
- Classes that we dont want public visibility should not be visible in Java as well. Making it `public[sql]` still keeps it visible as a top-level in Java API docs. Either move the implementation inside the Trigger class, or move the `ProcessingTimeTrigger` into `sql/execution` package which is explicitly hidden from API docs.

My apologies for missing prior context and conversation. This is just my two cents. I am sure I am probably missing some aspect of the conversation.

Reply:
Could we define the boundary of the PR - I guess if we want to address @tdas suggestion, it is going to be more than https://github.com/HeartSaVioR/spark/commit/f8488cfb0a045806e125215b933fc10931b1b431 - just wanted to be clear how much we want to do in this PR.

Reply:
My suggestion is we follow the comment above by moving `OneTimeTrigger` too, and leave it there. That's more consistent, and consistent with the intent. This leaves the class potentially 'public' in the bytecode, but it was before, and that's true of lots of `private[spark]` classes anyway. While I wouldn't argue with further moving things to `sql/execution`, I think that just putting the implementations in their natural home right now sounds coherent and an improvement, and doesn't expand the change much.

Reply:
Sorry I guess I mixed up. My bad, not moving OneTimeTrigger but moving ContinuousTrigger. Let me enumerate necessary changes from what I understand:

* `OneTimeTrigger` is already in `Triggers.scala` but open to `public`, so need to add deprecation and plan to hide it.
  * I guess adding `private[sql]` would work since Triggers.scala is already in `sql.execution.streaming` package. Not sure exact package of `sql.execution` will be hidden.
  * Otherwise we can add replacement as same as what we do for ProcessingTime, but requires bigger changeset.
* Move `ContinuousTrigger` to `Triggers.scala` with `private[sql]` scope, and deprecate origin.

I guess both moving and deprecating make the changeset looking verbose, but I guess even in major release we may not want to remove public classes which haven't been deprecated.

I guess my commit (https://github.com/HeartSaVioR/spark/commit/f8488cfb0a045806e125215b933fc10931b1b431) mentioned above already covered it, so please take a look at the commit. If we are OK to go or would like to continue reviewing under the commit, I'll add the commit to the PR.

Reply:
Oops I misspoke, I mean "move the implementations to Triggers.scala". 
I'm personally OK with not even deprecating, just moving, as it's a major release and a small detail, but, OK with deprecation too.

Reply:
Yeah either is fine for me too. If we would like to have simpler one, skipping deprecation would work. If we would like to have safer one, deprecation would work. I'd like to ask the decision for committers/PMC members, as it seems like related to some policy on the project.

Reply:
To move this forward, I suggest we just move the class and skip deprecation. A note in the Spark 3.0 migration guide about streaming would be good, as we're removing a deprecated class anyway.